trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/TrinoS3ProxyResource.java

```diff
@@ -36,12 +34,14 @@ public class TrinoS3ProxyResource
 {
     private final SigningController signingController;
     private final TrinoS3ProxyClient proxyClient;
+    private final RestConfig restConfig;
 
     @Inject
-    public TrinoS3ProxyResource(SigningController signingController, TrinoS3ProxyClient proxyClient)
+    public TrinoS3ProxyResource(SigningController signingController, TrinoS3ProxyClient proxyClient, RestConfig restConfig)
     {
         this.signingController = requireNonNull(signingController, "signingController is null");
         this.proxyClient = requireNonNull(proxyClient, "proxyClient is null");
+        this.restConfig = requireNonNull(restConfig, "restConfig is null");
```

Comment:
Because config objects are mutable we don't use them directly. Instead, we assign the value from the config object to a immutable field.